### PR TITLE
setup gem user stamps

### PR DESCRIPTION
**Pull request by wangqiang1208**
Changes in this PR:

1. setup gem userstamp and models

2. using our own fork of userstamps and schema_plus

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -92,3 +92,7 @@ gem 'cancancan', '~> 1.9'
 
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
+
```
Comment (lowjoel) [inline Gemfile]: once again this can be put together with the gems at the top, after migration comments, I think.

File: app/models/course.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 class Course < ActiveRecord::Base
   acts_as_tenant(:instance)
+  # for automatic update userstamps
```
Comment (lowjoel) [inline app/models/course.rb]: I'd just remove this comment because practically every model would have this declaration.

**Comment by lowjoel** [comment]
After you fix both, can squash some of the duplicate commits, then it should be mergeable.

**Comment by wangqiang1208** [comment]
@lowjoel You've rebased the schema_plus to their lastest 2.0.0alpha ?  cannot run the tests using their latest code.

**Comment by lowjoel** [comment]
Don't use 2.0.0. That's for Rails 4.2

**Comment by wangqiang1208** [comment]
I've updated

**Comment by lowjoel** [comment]
@wangqiang1208 Please rebase, merge conflicts...

**Comment by wangqiang1208** [comment]
@lowjoel I've fixed the conflicts

**Pull request merged**